lib/obsolete.gd

```diff
@@ -76,9 +76,9 @@
 ##  <#/GAPDoc>
 ##
 
-BIND_GLOBAL( "DeclareObsoleteSynonym", function( name_obsolete, name_current, desc )
+BIND_GLOBAL( "DeclareObsoleteSynonym", function( name_obsolete, name_current )
```

Comment:
It's not just |not guaranteed", it is completely unclear whether it'll ever happen. If we leave a value in, somebody has to take care of updating it before it becomes obsolete -- which clearly here did not happen.

And as long as these are used by lots of packages, realistically we are not going to make them go away.

If we ever have a concrete plan for phasing out obsolete identifiers, we can still re-introduce warnings referencing specific versions -- *if* that'll be necessary by that hypothetical plan.